lib/modules/commentNavigator.js

```diff
@@ -286,7 +302,6 @@ export function getPostsByCategory() {
 		if (!_posts[currentCategory] || currentCategory === 'highlighted') {
```

Comment:
This this function now is only called when the selection is changed, it can be simplified to `_posts[currentCategory] = sortTypes[currentCategory] ? sortTypes[currentCategory].getPosts() : sortTypes[currentCategory];` and renamed to something more fitting like `refreshCurrentCategoryPosts`.

Reply:
I don't get why we'd assign `_posts[currentCategory] = ... : sortTypes[currentCategory]`.

Reply:
Sorry, that was a typo. Anyway, now that the scroll is suppressed when opening the commentNav by default, we may as well remove the empty select option as that way remove a lot of logic related to that.

Reply:
Can you correct the typo? Did you mean to set it to a reasonable default like `_posts[currentCategory] = sortTypes[currentCategory] ? sortTypes[currentCategory].getPosts() : {};`?

Also, something causes the scroll to only be suppressed when "IAmA" is pre-set. I removed the empty option and navigated to a non-AMA page. "submitter" was selected, and the page did jump. I could dig deeper and clean it up, but could we do that as a separate PR? It's quite a bit of refactoring, and I'd have to understand the module better to really test it. 

Also, now that I think about it, what would you like to default the option to, if I were to remove the empty option?

Reply:
I was thinking `_posts[currentCategory] = sortTypes[currentCategory] ? sortTypes[currentCategory].getPosts() : [];` as `sortTypes[currentCategory]` is not defined for `''`, and `getPosts()` return `Arrray<HTMLElement>`. Anyway, the last commit appears to have made everything work, so my suggestion can be disregarded :)